recipe/build.sh

```diff
@@ -78,6 +78,9 @@ if [[ $(uname) == Darwin ]]; then
   export LDFLAGS="-headerpad_max_install_names -Wl,-no_compact_unwind $LDFLAGS"
   export ESMF_F90LINKOPTS="$LDFLAGS -pthread -lc++"
   export ESMF_CXXLINKOPTS="$LDFLAGS -pthread"
+else
+  export ESMF_F90LINKOPTS="$LDFLAGS"
+  export ESMF_CXXLINKOPTS="$LDFLAGS"
```

Comment:
@minrk, if you have time, could you confirm that this is what you had in mind in https://github.com/conda-forge/openmpi-feedstock/issues/143#issuecomment-2160197969?

Reply:
Yup, exactly what I meant. Thanks!

Though it looks like there's also [ESMF_CLINKOPTS](https://github.com/esmf-org/esmf/blob/13be2f7f6d94ba56a5cf3d3a831cb18cd7818279/makefile#L394) because a C file is failing to link.

Maybe there should be a suggestion to upstream to initialize all the LINKOPTS from $LDFLAGS? That seems a sensible default to me.

Reply:
Okay I just added that as well.  Yes, I think that would be a sensible fix upstream.  I'll suggest that when I have a bit more time.